libraries/AP_InertialSensor/AP_InertialSensor_Invensense.cpp

```diff
@@ -547,11 +590,16 @@ bool AP_InertialSensor_Invensense::_accumulate(uint8_t *samples, uint8_t n_sampl
 
         int16_t t2 = int16_val(data, 3);
         if (!_check_raw_temp(t2)) {
-            if (!hal.scheduler->in_expected_delay()) {
-                debug("temp reset IMU[%u] %d %d", _accel_instance, _raw_temp, t2);
+            if (_enable_fast_fifo_reset) {
+                _fast_fifo_reset();
+                return true;
```

Comment:
You are still going to lose data from the fifo surely so this needs to return false?